crates/syntax/src/ast/edit_in_place.rs

```diff
@@ -93,21 +93,16 @@ impl GenericParamsOwnerEdit for ast::Enum {
             } else {
                 Position::last_child_of(self.syntax().clone())
             };
-            create_where_clause(position, true)
+            create_where_clause(position)
         }
         self.where_clause().unwrap()
     }
 }
 
-fn create_where_clause(position: Position, after: bool) {
-    let mut elements = vec![make::where_clause(empty()).clone_for_update().syntax().clone().into()];
-    let ws = make::tokens::single_space().into();
-    if after {
-        elements.insert(0, ws)
-    } else {
-        elements.push(ws)
-    }
-    ted::insert_all(position, elements);
+fn create_where_clause(position: Position) {
+    let where_clause: SyntaxElement =
+        make::where_clause(empty()).clone_for_update().syntax().clone().into();
+    ted::insert_ws(position, where_clause);
 }
```

Comment:
I've been dreaming about the API which can magically fix up whitespace for years, super satisfying to just finally have it!

Reply:
What about matching brackets insertion, would that be possible to have?